proposals/auto-upgrade-critical-vulnerabilities.md

```diff
@@ -0,0 +1,132 @@
+
+1. Upgrades might contain breaking changes, despite developer claims. These might be design changes or bugs. Amelioration: automated updates will only begin after a gray period in which the changes are available for discretionary updates.
+
+2. Upgrades might be attacked by insiders within the auto-upgrade system, whether motivated by personal goals or coercion. Amelioration: at least three people should approve an auto-upgrade. To prevent nation-state coercion, the approvers must span more than one country.
+
+3. Technical vulnerabilities in upgrade infrastructure might be used to trigger malicious upgrades. Amelioration: security practices are good enough to bear the responsibility.
+
+#### Operations
+
+There is an organization or group administering the project. 
+
+There is a website for submitting, maintaining, and fetching auto-updates.
+
+- Submitters have a UI to create and edit an alert. They are most likely a maintainer of the affected package, but they might also be a security researcher or an interested third party. In some cases, an administrator might submit an alert.
+- Administrators maintain alerts. They can edit metadata. They can publish or reject an alert. They can also message submitters to bring an alert to a publishable state. In a separate line of responsibility, they do development on the site.
+- Fetchers are probably users of a package manager checking for alerts. They can match the packages they use with active alerts. Their main point of contact is a REST API.
+- Package consumers are users of a package manager investigating an auto-update.
+
+#### Go-To-Market and Roadmap
+
+On the demand side, the main distribution channel for upgrades is package management tooling. To ease the bottleneck, the auto-update project will contribute as much of the package management labor as possible. It will work closely with maintainers of package management tooling such as NPM, submitting PRs for necessary code changes in CLIs and websites.
+
+On the supply side, the bottleneck is learning about relevant patches. Maintainers of the auto-update project will develop relationships with the CVE community, raising awareness with articles and conference talks.
+
+Roadmap:
+1. Sign-off and funding
+2. Website for creating and publishing auto-upgrades
+3. REST API for fetching auto-upgrades
+4. Implementation in the first package manager
+5. First production auto-upgrade
+6. Iterate through most common package managers
+7. Widely available and in use
+
+### Due diligence
+
+<!-- Please answer the following due diligence questions; it's okay to answer "N/A" if you don't know yet. -->
+
+1. **What's your theory of change -- what impact do you expect this project to have? What's its goal?** <!-- Insert answer here -->
+  * Critical vulnerabilities will be flushed out of the open-source ecosystem much more quickly, leaving the Internet more secure. 
+2. **What related work has already been done in this area?** <!-- Insert answer here --> 
```

Comment:
Debian `unattended-upgrades` seems like a proof point, though with a different target. Because it includes dependency updates (though typically at system rather than language package manager level), and mostly because it involves independent curation (by the Debian security team IIRC) which presumably also takes into account the risk being mitigated and incurred by adding a security update which will be picked up by machines running unattended-upgrades. Sort of halfway between the idea here of automated upgrading of language package level dependencies based on an independent vetting (this proposal, IIUC) and the ubiquity of automated upgrading of apps based based on the app developers' process.